query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -65,7 +69,10 @@
     // how it is created.
     for (Expression expr : allSelections) {
       pqlBuilder.append(delim);
-      pqlBuilder.append(convertExpression2String(expr, paramsBuilder, executionContext));
+      pqlBuilder.append(
+          isGroupingSelectionForMapAttribute(expr, request.getGroupByList())
+              ? convertExpression2StringForMapAttribute(expr, paramsBuilder, executionContext)
```

Comment:
`convertExpression2StringForMapAttribute` is internally falling back to `convertExpression2String`, do we need the check here?

Reply:
I have added a check in this function only. Can move it up where this function is called. Should I move it there only ?

Reply:
Discussing this on the other thread.

Reply:
resolved